Use `datetime-local` instead of datetime picker in event settings

SudharakaP opened the pull request:
Resolve https://github.com/opencollective/opencollective/issues/4865

![event-creation](https://user-images.githubusercontent.com/12435965/144289915-da9e873a-2a9b-42dc-8272-0f43c51732b1.gif)

Betree commented:
Do we still need `react-datetime`?

Betree commented:
I get this when I try to edit the start date for any event, with any value

![image](https://user-images.githubusercontent.com/1556356/144404878-89b1bbe9-8a4b-43f9-a08f-a854a8e27f75.png)

znarf commented:
On my computer it works, but the end result doesn't match the specs. The stored date is not the date passed. Certainly related to Ben's issue.

**The date you see in the PostgresSQL row should be the one the user inputed and originally passed.**

znarf commented:
I added a commit which seems to take care of this for the Edit part.

znarf commented:
Added more commits because the behavior was not making much sense.

znarf commented on components/EditEventForm.js:
```diff
@@ -73,23 +74,10 @@ class EditEventForm extends React.Component {
 
-    // Make sure that endsAt is always >= startsAt
     if (fieldname === 'startsAt') {
-      const endsAt = this.state.event.endsAt;
-      if (!endsAt || new Date(endsAt) < new Date(value)) {
-        let newEndDate = new Date(value);
-        if (!endsAt) {
-          newEndDate.setHours(newEndDate.getHours() + 2);
-        } else {
-          // https://github.com/opencollective/opencollective/issues/1232
-          const endsAtDate = new Date(endsAt);
-          newEndDate = new Date(value);
-          newEndDate.setHours(endsAtDate.getHours());
-          newEndDate.setMinutes(endsAtDate.getMinutes());
-        }
-        value = newEndDate.toString();
-        event['endsAt'] = value;
-      }
```
Just leave endsAt alone and no magic crazyness.

znarf commented on components/EditEventForm.js:
```diff
@@ -120,9 +116,9 @@ class EditEventForm extends React.Component {
-    defaultStartsAt.setHours(19);
-    defaultStartsAt.setMinutes(0);
+
+    const defaultStartsAt = dayjs().utc().set('hour', 19).set('minute', 0);
+    const defaultEndsAt = dayjs().utc().set('hour', 22).set('minute', 0);
```
Use dayjs and utcise.

znarf commented on components/edit-collective/Form.js:
```diff
@@ -327,7 +328,7 @@ class EditCollectiveForm extends React.Component {
         set(collective, 'settings.applyMessage', value);
       } else if (fieldname === 'startsAt' && collective.type === EVENT) {
-        collective[fieldname] = value;
+        collective[fieldname] = dayjs(value).format('YYYY-MM-DD HH:mm:ss+00');
```
This is the format expected by the GraphQL API v1 (Postgres)

znarf commented on components/edit-collective/Form.js:
```diff
@@ -649,10 +652,14 @@ class EditCollectiveForm extends React.Component {
-    defaultStartsAt.setHours(19);
-    defaultStartsAt.setMinutes(0);
+
+    const defaultStartsAt = dayjs().utc().set('hour', 19).set('minute', 0);
+    const defaultEndsAt = dayjs(collective.startsAt || defaultStartsAt)
+      .utc()
+      .set('hour', 22)
+      .set('minute', 0);
```
Use dayjs

SudharakaP commented:
> Do we still need `react-datetime`?

@Betree : I don't think so, but I need to carefully check this one. So I've created a separate issue so that we can remove this; https://github.com/opencollective/opencollective/issues/5010

SudharakaP commented:
@Betree @znarf : Thanks for the corrections. I've added some small things here; 

1) In our old date time picker it automatically had support for minimum date time (cannot pick a date before today). I've added this to the `datetime-local` as well by specifying the `min` attribute. 

2) Depending on the browser the `datetime-local` input might allow the user to specify nothing (Firefox in particular). And if they do that we probably need to have a nicer error message rather than the one @Betree saw above. I've added this at, https://github.com/opencollective/opencollective-api/pull/6915

SudharakaP commented:
Seems a e2e test is failing after the latest commit. Looking into this. 🤔

SudharakaP commented:
> Seems a e2e test is failing after the latest commit. Looking into this. thinking

This is due to my API PR; https://github.com/opencollective/opencollective-api/pull/6915. This problem isn't specific to this issue so therefore I'll tackle this on a separate issue at, https://github.com/opencollective/opencollective/issues/5011
